kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -251,6 +251,15 @@ private static StructType convertToPhysicalSchema(
       StructType prunedSchema, StructType fullSchema, boolean includeFieldId) {
     StructType newSchema = new StructType();
     for (StructField prunedField : prunedSchema.fields()) {
+      if (fullSchema.indexOf(prunedField.getName()) == -1) {
```

Comment:
Do we want to hard check the column is a valid metadata column checkState()? (both name and type matches the available metadata column)

Reply:
You're right. I initially chose this kind of "hot fix" because the current main branch would also throw a NullPointerException at this point for unknown columns.

I'll look at your pointer tomorrow. We might want to address it once we agreed on one of the proposals for a metadata column API so that we can address regular and metadata columns holistically.